src/main/java/use_case/delete_post/DeletePostInteractor.java

```diff
@@ -27,30 +27,45 @@ public void deletePost(DeletePostInputData deletePostInputData) {
             throw new DeletePostFailedException("Post with given ID doesn't exist.");
         }
 
-
-        boolean userCanDelete = canDelete(deletePostInputData);
-        if (!userCanDelete) {
+        if (!canDelete(deletePostInputData)) {
             postPresenter.prepareFailView("User does not have permission to delete this post.");
             throw new DeletePostFailedException("User does not have permission to delete this post.");
         }
 
         try {
             postDataAccessObject.deletePost(deletePostInputData.getPostId());
+            userRepo.removePostFromUser(deletePostInputData.getCurrentUserId(), deletePostInputData.getPostId());
 
-            DeletePostOutputData outputData = new DeletePostOutputData(
-                    deletePostInputData.getPostId(),
-                    true
-            );
+            final DeletePostOutputData outputData = new DeletePostOutputData(deletePostInputData.getPostId(), true);
             postPresenter.prepareSuccessView(outputData);
         } catch (Exception e) {
             postPresenter.prepareFailView("Failed to delete the post.");
             throw new DeletePostFailedException("Failed to delete the post.");
         }
     }
 
-    public boolean canDelete(DeletePostInputData post) {
-        final User currentUser = userRepo.getCurrentUser();
-        return currentUser.getUserID().equals(post.getUserId()) ||
-                currentUser.getModerating().contains(post.getPostId());
+    /**
+     * Check if the user can delete the post.
+     * @param inputData the input data
+     * @return true if the user can delete the post, false otherwise
+     */
+    public boolean canDelete(DeletePostInputData inputData) {
```

Comment:
I think it would quite confusing for developers in the future to look at inputData and not know what's in it - you should have the parameters take in whatever is needed to execute the canDelete method, like the postId

also this method should be private, when you test this use case you should be testing on the deletePost method, and your test cases should cover if user can delete or not